cristin-organization/src/test/java/no/unit/nva/cristin/organization/FetchCristinOrganizationHandlerTest.java

```diff
@@ -43,54 +40,56 @@
 import static no.unit.nva.testutils.RandomDataGenerator.randomString;
 import static no.unit.nva.utils.UriUtils.getCristinUri;
 import static no.unit.nva.utils.UriUtils.getNvaApiId;
-import static nva.commons.apigateway.ApiGatewayHandler.MESSAGE_FOR_RUNTIME_EXCEPTIONS_HIDING_IMPLEMENTATION_DETAILS_TO_API_CLIENTS;
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.hasKey;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.when;
 
 class FetchCristinOrganizationHandlerTest {
 
     public static final ObjectMapper restApiMapper = JsonUtils.dtoObjectMapper;
     public static final String IDENTIFIER = "identifier";
     public static final String NON_EXISTING_IDENTIFIER = "1.0.0.0";
-    public static final String ORGANIZATION_NOT_FOUND_MESSAGE = "cannot be dereferenced";
     public static final String LIST_OF_UNITS_JSON_FILE = "list_of_unit_185_53_18_14.json";
     public static final String CRISTIN_UNITS_URI = "https://api.cristin-test.uio.no/v2/units?id=185.53.18.14";
     public static final String ENGLISH_LANGUAGE_KEY = "en";
     public static final String DEPARTMENT_OF_MEDICAL_BIOCHEMISTRY = "Department of Medical Biochemistry";
+    public static final String EMPTY_JSON = "{}";
 
     private FetchCristinOrganizationHandler fetchCristinOrganizationHandler;
     private CristinOrganizationApiClient cristinApiClient;
+    private DefaultOrgFetchProvider clientProvider;
     private ByteArrayOutputStream output;
     private Context context;
-    private HttpClient mockHttpClient;
 
     @BeforeEach
     void setUp() {
         context = mock(Context.class);
-        mockHttpClient = mock(HttpClient.class);
+        HttpClient mockHttpClient = mock(HttpClient.class);
         cristinApiClient = new CristinOrganizationApiClient(mockHttpClient);
+        cristinApiClient = spy(cristinApiClient);
+        doReturn(Try.of(new HttpResponseFaker(EMPTY_JSON)))
+            .when(cristinApiClient).sendRequestMultipleTimes(any());
```

Comment:
I don't like the method name `sendRequestMultipleTimes()`. Either use just `sendRequest()`, or if it is important that it does retry, consider `sendRequestWithRetry()`. `sendRequestMultipleTimes()` indicates to me that it always send the request multiple times.

Reply:
sendRequestWithRetry

Reply:
This is just for compatibility with existing code. in the new version i will use another method